api/v1/bandwidth.go

```diff
@@ -0,0 +1,67 @@
+package api
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"github.com/celestiaorg/bittwister/xdp/bandwidth"
+	"go.uber.org/zap"
+)
+
+// BandwidthStart implements POST /bandwidth/start
+func (a *RESTApiV1) BandwidthStart(resp http.ResponseWriter, req *http.Request) {
+	var body BandwidthStartRequest
+	if err := json.NewDecoder(req.Body).Decode(&body); err != nil {
+		sendJSONError(resp,
+			MetaMessage{
+				Type:    APIMetaMessageTypeError,
+				Slug:    SlugJSONDecodeFailed,
+				Title:   "JSON decode failed",
+				Message: err.Error(),
+			},
+			http.StatusBadRequest)
+		return
+	}
+
+	if a.bw == nil {
+		a.bw = &netRestrictService{
+			service: &bandwidth.Bandwidth{
+				Limit: body.Limit,
+			},
+			logger: a.logger,
+		}
+	} else {
+		bw, ok := a.bw.service.(*bandwidth.Bandwidth)
+		if !ok {
+			sendJSONError(resp,
+				MetaMessage{
+					Type:    APIMetaMessageTypeError,
+					Slug:    SlugTypeError,
+					Title:   "Type cast error",
+					Message: "could not cast netRestrictService.service to *bandwidth.Bandwidth",
+				},
+				http.StatusInternalServerError)
+			return
+		}
+		bw.Limit = body.Limit
```

Comment:
how is this like being used? `bw` is scoped to this else condition so this assignment is a no-op right?